### ISSUE TITLE
Enable auto requestPointerLock()

### ISSUE THREAD

**Issue by IceReaper** [nwjs/nw.js#5473]
The current implementation relies on requestPointerLock only being trigger-able when the user interacts with a click. While this is a good idea in general for browsers, it causes some struggling with nw.js based first-person controlled applications. When switching windows, the focus is lost, and so the pointerlock released. It cannot be automatically re-attached without an explicit click.

nw.js should be able to bypass this behavior using a `allowAutoPointerLock: true` package.json property. Its pretty annoying if you switch back to the application (a first-person-shooter in my case) and you can start to walk around with the keyboard, but the mouse cursor is still lost. This is also an inconsistency with the Gamepads, which work without the need of clicking.

In theory this piece of code should work out of the box:

    InputManager.prototype.step = function () {
        if (this.mouselock && document.pointerLockElement !== document.body) {
            document.body.requestPointerLock();
        } else if (!this.mouselock && document.pointerLockElement === document.body) {
            document.exitPointerLock();
        }
    };

When opening the menu, the mouselock boolean is set to false. also this is automatically set to true after the loading screen has been completed and the game is entered. The need of an additional click at this point plus an additional click when switching windows and back is just a pain and makes the application feel poorly written, hence the request of the property mentioned above.

**Comment by georgir** [comment]
Have you tested doing it in the window's focus event handler?

**Comment by IceReaper** [comment]
Yup, basicaly any event handler works. However if you trigger it by a network packet via socket or a finished loading screen or when clicking on "buy" in the weapons-shop for example, it wont work, as the actual "bought, close window now and continue" might be triggered in the end by a websocket response packet. I could come up with some more cases in which it might be required to allow to switch the mouse behavior directly without user interaction. If declined, no problem, then i need to implement an overlay "please click" or such. However as said above, that somehow does not feel right for some cases.
